lib/form-layout: Add support for widgets nested in splits

garrett opened the pull request:
Fix alignment as mentioned in https://github.com/cockpit-project/cockpit/pull/11083/#issuecomment-460710561 by applying special widget offseting & min-height to widgets also directly enclosed within `ct-form-layout-split`, which should behave now similarly to top-level children.

garrett commented:
Ideally, the widget would be a direct child of the grid element. That's what splits are designed for. However, in looking at the code, it would be difficult to "fix" the HTML to work as intended, so I basically adjusted the CSS to account for cases like this, as it'll probably show up again in the future too (and isn't really "wrong").

garrett commented:
Oh, this will probably clash with https://github.com/cockpit-project/cockpit/pull/11035/commits/b384c4b50a5624a99b800b430d55bf73bcaab365, which hasn't been merged yet. That specific commit is similar and we can add this same logic to https://github.com/cockpit-project/cockpit/pull/11035/commits/b384c4b50a5624a99b800b430d55bf73bcaab365#diff-065d8dbbc10bd4f648805392b2b935f6R137 (line 137)

garrett commented:
Merged the two commits into one.

andreasn commented:
The text is still one pixel off, but maybe that's a general problem with checkboxes? Look a lot better with this PR applied, in general, though.

![screenshot from 2019-02-11 17-59-52](https://user-images.githubusercontent.com/209895/52579788-33d96c00-2e27-11e9-91ea-4a24e174e6b3.png)

garrett commented:
> The text is still one pixel off

Oh, hmm. I should investigate this. :wink:

garrett commented:
Fixed checkboxes and radios in various scenarios and added list-group-item fixes here now too. (Network page among others had odd stuff going on. I had some local fixes in another PR, but it really belongs here.)

garrett commented:
I need to do further testing tomorrow and perhaps some little fixups.

garrett commented:
The changes here and in #11035 don't work well together. I'll have to investigate further to make everything work. It might be local to that PR, or perhaps I'm making wrong assumptions somewhere. :unamused:

martinpitt commented:
Sorry, this fell through the cracks. This needs to be rebased.

garrett commented:
That's OK: I'm not quite done with it either, if I remember correctly. I got pulled over to other PRs, interacting with other people.

I'm also pretty certain I did some commit collisions with myself. :wink:

garrett commented:
Rebased, as to not let this get too stale; need to check (and fix) it everywhere still.

garrett commented:
This merged version still has a few errors:

![Screenshot_2019-03-14 Storage - Sunny](https://user-images.githubusercontent.com/10246/54375838-14ca3600-4682-11e9-86f9-5a3befdec041.png)

![Screenshot_2019-03-14 Storage - Sunny(1)](https://user-images.githubusercontent.com/10246/54375863-214e8e80-4682-11e9-8a71-fde349dc6e30.png)

And things are probably off a pixel or two elsewhere (checkboxes, most likely). Still needs work.

garrett commented:
I really need to find the time to finish this and de-conflict-ify it (again), then fix it up again. We still have a few bugs in Cockpit that are addressed in this PR.

marusak commented:
Needs to rebase to master since #12367 changed tests names

garrett commented:
I think I basically implemented this either during the PF4-style refresh or shortly before.
